data-prepper-plugins/mutate-event-processors/README.md

```diff
@@ -0,0 +1,209 @@
+  * `overwrite_if_key_exists` - (optional) - When set to `true`, if `key` already exists in the event, then the existing value will be overwritten. The default is `false`. 
+
+___
+
+##CopyValueProcessor
+A processor that copies values within an event
+
+###Basic Usage
+To get started, create the following `pipeline.yaml`.
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - copy_value:
+        entries:
+        - from_key: "message"
+          to_key: "newMessage"
+          overwrite_if_to_key_exists: true
+  sink:
+    - stdout:
+```
+
+Create the following file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the path of this file.
+
+```json
+{"message": "value"}
+```
+
+When run, the processor will parse the message into the following output:
+
+```json
+{"message": "value", "newMessage": "value"}
+```
+
+> If `newMessage` had already existed, its existing value would have been overwritten with `value`
+
+###Configuration
+* `entries` - (required) - A list of entries to be copied in an event
+    * `from_key` - (required) - The key of the entry to be copied
+    * `to_key` - (required) - The key of the new entry to be added
+    * `overwrite_if_to_key_exists` - (optional) - When set to `true`, if `to_key` already exists in the event, then the existing value will be overwritten. The default is `false`.
+
+___
+
+##DeleteEntryProcessor
+A processor that deletes entries in an event
+
+###Basic Usage
+To get started, create the following `pipeline.yaml`.
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - delete_entry:
+        entries:
+        - with_key: "message"
+  sink:
+    - stdout:
+```
+
+Create the following file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the path of this file.
+
+```json
+{"message": "value", "message2": "value2"}
+```
+
+When run, the processor will parse the message into the following output:
+
+```json
+{"message2": "value2"}
+```
+
+> If `message` had not existed in the event, then nothing would have happened
+
+###Configuration
+* `entries` - (required) - A list of entries to be deleted from an event
+    * `with_key` - (required) - The key of the entry to be deleted
+
+___
+
+##RenameKeyProcessor
+A processor that renames keys in an event
+
+###Basic Usage
+To get started, create the following `pipeline.yaml`.
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - rename_key:
+        entries:
+        - from_key: "message"
+          to_key: "newMessage"
+          overwrite_if_to_key_exists: true
```

Comment:
Just want to clarify. Given this configuration with the following Event:

```
{ "message": "my_message", "new_message": "my_new_message" }
```

what happens to the original `new_message` if `overwrite_if_to_key_exists` is `true`? Does it remove that entry completely and have this result?

```
{ "new_message": "my_message" }
```